### PR TITLE
feat(server): add base server with health checks

### PR DESCRIPTION
**Pull request by harrisonlob**
**Why:** To ramp up GO knowledge

**What:** Creates a basic server in Go (based on Goyagi)

**Details:**

**Notes:**

### CODE REVIEW

File: cmd/serve/main.go [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+	log.Info("server started")
+
+	err := srv.ListenAndServe()
+	if err != nil && err != http.ErrServerClosed {
+		log.Err(err).Fatal("server stopped")
+	}
+
+	log.Info("server stopped")
+}
```
Comment (dmlittle) [inline cmd/serve/main.go]: This file (and a few others) don't have a newline at the end. GitHub shows this with the following symbol:
image

A line in a file is defined as a sequence of non-newline characters terminated by a newline (`\n` byte). While most systems have designed around people forgetting to include it, it is good practice to add them. You can actually configure your editor to automatically add newlines at the end of your files.

In VSCode you can do this by adding the following to your configs:
```json
{
    "files.insertFinalNewline": true,
}
```

File: pkg/server/server.go [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+	"github.com/harrisonlob/goyagi/pkg/signals"
+)
+
+// New returns a new HTTP server with the registered routes.
+func New() *http.Server {
+	log := logger.New()
+
+	e := echo.New()
+
+	health.RegisterRoutes(e)
```
Comment (dmlittle) [inline pkg/server/server.go:20]: Supernit (optional): Usually I like to separate different logical sections with a blank line. In this particular case I would probably add a new line in between registering the health routes and creating the http server struct.

File: pkg/server/server.go [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+	srv:= &http.Server {
```
Comment (dmlittle) [inline pkg/server/server.go]: When instantiating a struct there shouldn't be a space in between the struct type and the opening brace (`{`).

In VSCode you can have the Go plugin automatically format your code and fix any styling issues automatically when you save a file by enabling the following configuration:
```json
{
    "go.formatTool": "goimports",
}
```

**Review by dmlittle — COMMENTED** [review]
Left a few comments! The code looks good save a few minor syntax / style issues.

File: cmd/serve/main.go (hunk 1 above)
Comment (harrisonlob) [reply cmd/serve/main.go]: Thanks for the heads up!

**Review by dmlittle — APPROVED** [review]

File: pkg/server/server.go (hunk 2 above)
Comment (virtualdom) [reply pkg/server/server.go:20]: I agree with Donald -- let's add a new line after this 🙏

**Review by virtualdom — COMMENTED** [review]
Looks good, but I'll approve after Donald's nit is resolved!

File: pkg/server/server.go [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+)
+
+// New returns a new HTTP server with the registered routes.
+func New() *http.Server {
+	log := logger.New()
+
+	e := echo.New()
+
+	health.RegisterRoutes(e)
+	srv:= &http.Server{
```
Comment (virtualdom) [inline pkg/server/server.go]: Let's add a space before the `:=` here too!

Comment (harrisonlob) [reply pkg/server/server.go]: @virtualdom fixed!

File: pkg/server/server.go (hunk 2 above)
Comment (harrisonlob) [reply pkg/server/server.go:20]: @dmlittle @virtualdom - this should be fixed in latest commit as well

**Review by virtualdom — APPROVED** [review]

**Pull request merged**